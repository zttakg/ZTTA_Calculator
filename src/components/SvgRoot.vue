<template>
  <svg
    :width="width"
    :height="height"
    :viewBox="viewBox"
    xmlns="http://www.w3.org/2000/svg"
    version="1.1"
    xmlns:xlink="http://www.w3.org/1999/xlink">
    <slot></slot>
  </svg>
</template>

<script>
export default {
  name: "SvgRoot",

  props: {
    width: {
      type: Number,
      default: 500
    },
    height: {
      type: Number,
      default: 500
    },
    viewBox: {
      type: String,
      default() {
        return `0 0 ${this.width} ${this.height}`;
      }
    }
  }
};
</script>
