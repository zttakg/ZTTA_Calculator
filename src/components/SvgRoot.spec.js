import { shallowMount } from "@vue/test-utils";
import SvgRoot from "./SvgRoot";

describe("<SvgRoot /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(SvgRoot);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(SvgRoot, {
      propsData: {
        width: 100,
        height: 200,
        viewBox: "0 0 500 800"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
