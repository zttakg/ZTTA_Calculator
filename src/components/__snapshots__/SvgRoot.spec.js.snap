// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgRoot /> component should render all props 1`] = `<svg width="100" height="200" viewBox="0 0 500 800" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink"></svg>`;

exports[`<SvgRoot /> component should render correct without props 1`] = `<svg width="500" height="500" viewBox="0 0 500 500" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink"></svg>`;
